src/main/webapp/features/shared/services/game/Game.service.js

```diff
@@ -0,0 +1,38 @@
+'use strict';
+
+angular.module('catan')
+        .factory('Game', ['GameModel', '$q', 'Remote', function (GameModel, $q, Remote) {
+
+            var Game = function(idOrDetails) {
+                return new GameModel(idOrDetails);
+            };
+
+            Game.findPublic = function() {
+                return this.findByRemoteService("game", "listPublic");
+            };
+
+            Game.findCurrent = function() {
+                return this.findByRemoteService("game", "listCurrent");
+            };
+
+            Game.findByRemoteService = function(remoteGroup, remoteRequest) {
```

Comment:
we can leave [ct-game-list type-of-games="CURRENT"], after that in ct-games-list.js, you should pass scope.typeOfGames directly in the method, that you have to create in Game service,for example Game.getListOfGames(scope.typeOfGames); so that you remove redundant logic in ct-games-list.js and redundant methods in Game.service.js, but lets discuss it in skype if my explanation is not clear